### ISSUE TITLE
sift extract with less keypoints for megadepth datasets

### ISSUE THREAD

**Issue by AbyssGaze** [magicleap/SuperGluePretrainedNetwork#70]
Thanks for your amazing work. 
I have implemented superpoint extact superglue on megadepth with some  imperfection(depth loss when interpolate, I will discard that image pair).
![image](https://user-images.githubusercontent.com/14217667/111154938-bcaacf00-85ce-11eb-91e5-6e99cc589dc1.png)
The submission in https://www.visuallocalization.net/ is :
![visualization](https://user-images.githubusercontent.com/14217667/111154620-6342a000-85ce-11eb-938b-a7ec19906225.png)
My implementation with online keypoints extraction, if superpoints less than 1024/2048 I will random select points with interpolation descriptors, but when migrating to **sift**, some images with too little keypoints, how to deal that problem?
Thanks

**Comment by Skydes** [comment]
Great job, this looks pretty good! For SIFT we randomly sampled points but also descriptors, as the absolute value of a Gaussian in a D256 space, potentially with root normalization to emulate RootSIFT.

**Comment by AbyssGaze** [comment]
Thanks awfully!
I will try it and public RootSIFT results.

**Comment by zenmedou** [comment]
> Thanks for your amazing work.
> I have implemented superpoint extact superglue on megadepth with some imperfection(depth loss when interpolate, I will discard that image pair).
> ![image](https://user-images.githubusercontent.com/14217667/111154938-bcaacf00-85ce-11eb-91e5-6e99cc589dc1.png)
> The submission in https://www.visuallocalization.net/ is :
> ![visualization](https://user-images.githubusercontent.com/14217667/111154620-6342a000-85ce-11eb-938b-a7ec19906225.png)
> My implementation with online keypoints extraction, if superpoints less than 1024/2048 I will random select points with interpolation descriptors, but when migrating to **sift**, some images with too little keypoints, how to deal that problem?
> Thanks

Hi，this is really high performance! May I ask if it is combined with hloc or not? Thanks.

**Comment by AbyssGaze** [comment]
Yes, It's combined with hloc. I just have reproduced Sarlin's work(superglue with megadepth).

**Comment by AbyssGaze** [comment]
@Skydes Thanks for your suggestion, I have used rootsift. Although the loss decreases and the matching performance get  better, the loss decreases to 1.0 and it converges. Does sift(DOG) keypoints need to do nms as superpoint? Thanks!
![14_400_matches](https://user-images.githubusercontent.com/14217667/112408523-e7c1bb00-8d52-11eb-9d1c-8d7b4363ba17.png)

**Comment by zenmedou** [comment]
HI, thanks for the reply. May I ask what if hloc is not added? I'm also trying to reproduce results on megadepth but have some performance gap for sg+sp. Approximately 77 vs 79(official model) on aachen. I'm using offline data generation which yields about 600k pairs in total.

**Comment by Skydes** [comment]
@AbyssGaze DoG does not need any NMS but we you need to define tighter thresholds when computing the ground truth, as keypoints are generally better localized. The exact value depends on your image resolution.

**Comment by szgy66** [comment]
> @Skydes Thanks for your suggestion, I have used rootsift. Although the loss decreases and the matching performance get better, the loss decreases to 1.0 and it converges. Does sift(DOG) keypoints need to do nms as superpoint? Thanks! ![14_400_matches](https://user-images.githubusercontent.com/14217667/112408523-e7c1bb00-8d52-11eb-9d1c-8d7b4363ba17.png)

hello,I want to ask that how combline sift with superglue? The length of the SIFT descriptor is 128D, but the superglue input is 256D? And does Superglue need to retrain? thanks!

**Comment by AbyssGaze** [comment]
Yes, we will publish the training codes no surprise on May.1.2022.

Yang Gao ***@***.***> 于2022年1月22日周六 16:19写道：

**Comment by yangfuwei** [comment]
Hi, 1-May-2022 have passed, Did you publish your training code as planned?

**Comment by AbyssGaze** [comment]
I'm very sorry, and now our open source process has become very complicated and demanding. We need to complete the OETR open-source process first, and we need to go through the magic of legal affairs. Suppose there are more papers published in the future. In that case, we will integrate them into the SuperRegistrator codebase (including code integration libraries such as superglue, bihome, and loftr) before starting the open-source process again. I am very sorry.
